ethereum/contracts/SwapCreator.sol

```diff
@@ -48,8 +48,8 @@ contract SwapCreator is Secp256k1 {
         Swap swap;
         // the fee, in wei, paid to the relayer
         uint256 fee;
-        // the address to be paid the relayer fee
-        address relayer;
+        // hash of (relayer's payout address || 4-byte salt)
+        bytes32 relayerHash;
```

Comment:
The randomized portion usually goes at the front of the hash. If someone wants to test a particular result against a list of addresses, they'll have to hash more bytes if the salt is first. It would need to be a large list for it to matter though.